app/src/main/java/unipd/se18/ocrcamera/cameraActiviti.java

```diff
@@ -0,0 +1,701 @@
+            });
+            return;
+        }
+
+        // Auto focus should be continuous for camera preview.
+        mCaptureRequestBuilder.set(CaptureRequest.CONTROL_MODE,
+                CaptureRequest.CONTROL_MODE_AUTO);
+        try
+        {
+            // Start the camera preview
+            mCaptureRequest = mCaptureRequestBuilder.build();
+            mCameraCaptureSession.setRepeatingRequest(mCaptureRequest, null, mBackgroundHandler);
+        }
+        catch (CameraAccessException e)
+        {
+            e.printStackTrace();
+        }
+
+    }
+
+    /**
+     *@param: communicate with the camera to have a picture
+     *@return: return the photo in a sharedPreferences
+     * Takes a photo.
+     * <p>Saves the captured photo, previously converted into Base64 String, into the current
+     * activity sharedPreferences</p>
+     * @modify mCameraDevice
+     * @author Alberto Valente, Taulant Bullaku
+     */
+    private void takePhoto() {
+
+        if(mCameraDevice == null)
+        {
+            Log.e("cameraDevice", "cameraDevice is null");
+            return;
+        }
+
+        CameraManager manager = (CameraManager) getSystemService(Context.CAMERA_SERVICE);
+
+        try
+        {
+            CameraCharacteristics characteristics = manager.getCameraCharacteristics(mCameraDevice.getId());
+            Size[] jpegSizes = Objects.requireNonNull(characteristics.get(
+                    CameraCharacteristics.SCALER_STREAM_CONFIGURATION_MAP)).getOutputSizes(ImageFormat.JPEG);
+            int width = 640;
+            int height = 480;
+            if(jpegSizes != null && jpegSizes.length > 0)
+            {
+                width = jpegSizes[0].getWidth();
+                height = jpegSizes[0].getHeight();
+            }
+
+            mImageReader = ImageReader.newInstance(width, height, ImageFormat.JPEG, 1);
+
+            ImageReader.OnImageAvailableListener readerListener = new ImageReader.OnImageAvailableListener()
+            {
+                @Override
+                public void onImageAvailable(ImageReader reader) {
+                    //acquires the last image and delivers it to a buffer
+                    Image image = reader.acquireLatestImage();
+                    ByteBuffer buffer = image.getPlanes()[0].getBuffer();
+                    byte[] photoByteArray = new byte[buffer.capacity()];
+                    buffer.get(photoByteArray);
+
+                    //Converts the byte array related to the given photo to a String in Base64 format
+                    String photoBitmapToString = Base64.encodeToString(photoByteArray, Base64.DEFAULT);
+
+                    //Create sharedPref file to save the Bitmap of last taken photo in a String form
+                    SharedPreferences sharedPref = getPreferences(Context.MODE_PRIVATE);
+                    SharedPreferences.Editor editor = sharedPref.edit();
+                    editor.putString("photoBitmap", photoBitmapToString);
```

Comment:
To avoid hard coding the key 'photoBitmap' you can save it using android String resources, it's very easy to use: https://developer.android.com/guide/topics/resources/string-resource#java